lib/libs/preferences.dart

```diff
@@ -173,7 +174,17 @@ class Preferences extends SerializableObject with ChangeNotifier {
   bool get stuPlanShowLastRoomUsage => attributes["sp_show_lru"] ?? true;
   set stuPlanShowLastRoomUsage(bool val) => setSaveNotify("sp_show_lru", val);
 
-  /// ungenutzt - eigentlich, damit der Benutzer Einträge in der Navigationsliste ausblenden kann
+  /// soll in der Detailansicht einer Stunde ein Link zum entsprechenden Raumplan angezeigt werden
+  bool get stuPlanShowRoomPlanLink => attributes["sp_show_rpl"] ?? true;
+  set stuPlanShowRoomPlanLink(bool val) => setSaveNotify("sp_show_rpl", val);
+
+  /// Filterliste für die anzuzeigenden Raumtypen
+  List<String> get filteredRoomTypes => cast<String>(attributes["filtered_room_types"])?.split(",") ?? listOfRoomTypeIdStrings();
+  set filteredRoomTypes(List<String> val) => setSaveNotify("filtered_room_types", val.join(","));
+  void addFilteredRoomType(RoomType id) => filteredRoomTypes = filteredRoomTypes..add(id.name);
+  void removeFilteredRoomType(RoomType id) => filteredRoomTypes = filteredRoomTypes..remove(id.name);
```

Comment:
```suggestion
  List<RoomType> get filteredRoomTypes => cast<String>(attributes["filtered_room_types"])?.split(",").map((val) => RoomType.values.firstWhere((t) => t.name == val, () => RoomType.unassigned)) ?? RoomType.values;
  set filteredRoomTypes(List<RoomType> val) => setSaveNotify("filtered_room_types", val.map((t) => t.name).join(","));
  void addFilteredRoomType(RoomType id) => filteredRoomTypes = filteredRoomTypes..add(id);
  void removeFilteredRoomType(RoomType id) => filteredRoomTypes = filteredRoomTypes..remove(id);
```
Durch die Erfassung und das Auslesen direkt als Enum hast du weniger Risiko, dass da mal Strings drinstehen die keinen Eintrag in RoomType darstellen (und ich finde es unangenehm, wenn man Enum-Werte als Strings übergibt).

Reply:
Ist auch sinnvoller so, ich hatte da viel probiert und war dann zu dem Schluss gekommen, dass das nicht gut funktioniert. Deswegen danke für deinen Hinweis. Ich habe das ungefähr so umgesetzt, mit einem zusätzlichen `toList`, da die Typen sonst nicht funktioniert hätten.